tests/ui_tools/test_boxes.py

```diff
@@ -340,6 +340,68 @@ def test_generic_autocomplete_mentions_subscribers(
         typeahead_string = write_box.generic_autocomplete(text, state)
         assert typeahead_string == required_typeahead
 
+    @pytest.mark.parametrize(
+        "text",
+        [
+            prefix + "Human"[: index + 1]
+            for index in range(len("Human"))
+            for prefix in ["@", "@**"]
+        ]
+        + ["@**"],
+    )
+    def test_generic_autocomplete_user_mentions(self, write_box, text, mocker, state=1):
+        _process_typeaheads = mocker.patch(BOXES + ".WriteBox._process_typeaheads")
+
+        matching_users = [
+            "Human Myself",
+            "Human 1",
+            "Human 2",
+            "Human Duplicate",
+            "Human Duplicate",
+        ]
+        distinct_matching_users = [
+            "@**Human Myself**",
+            "@**Human 1**",
+            "@**Human 2**",
+            "@**Human Duplicate|13**",
+            "@**Human Duplicate|14**",
+        ]
```

Comment:
If these are only specific to the assert part then they can sit with it, below the act[ion] part. That makes it clear they're not part of the setting-up.